pkg/cmd/pulumi/util.go

```diff
@@ -486,10 +486,7 @@ func chooseStack(ctx context.Context,
 // parseAndSaveConfigArray parses the config array and saves it as a config for
 // the provided stack.
 func parseAndSaveConfigArray(s backend.Stack, configArray []string, path bool) error {
-	if len(configArray) == 0 {
```

Comment:
Looks like this now _always_ calls saveConfig because the empty check in parseConfigArray still returns a map and nil-err.